docs/blog/posts/2024-11-14-pypi-now-supports-digital-attestations.md

```diff
@@ -0,0 +1,116 @@
+work to add support for [Trusted Publishing], as well as the [deprecation and
+removal of PGP signatures].
+
+<!-- more -->
+
+### Why not plain signatures?
+
+PyPI's support for digital attestations has three key advantages over regular
+cryptographic signatures, such as those provided by PGP:
+
+* **Attestations are signed by an identity, not a key pair:**
+  Similar to our recent support for Trusted Publishing, PyPI's support for
+  digital attestations relies upon Open ID Connect (OIDC) identities. By signing
+  attestations with identities, and not a public/private key pair, we mitigate the
+  potential for an individual's key loss or compromise, one of the most common failure cases for
+  PGP signing.
+* **Attestations provide a verifiable link to an upstream source repository:**
+  By signing with the identity of the upstream source repository, such as in the
+  case of an upload of a project built with GitHub Actions, PyPI's support for
+  digital attestations defines a strong and verifiable association between a file
+  on PyPI and the source repository, workflow, and even the commit hash that
+  produced and uploaded the file.
+* **Attestations are verified when uploaded, and must be verifiable to be uploaded:**
+  Upon review of the state of historical PGP signatures published to PyPI, [we
+  found that many signatures were not verifiable], either by PyPI or by end
+  users. With support for PEP 740, PyPI only permits attestations with a
+  verifiable signature to be uploaded and redistributed by the index. This
+  ensures that all attestations are verifiable and useful for all PyPI users.
+
+Much more detail is provided in a corresponding blog post by Trail of Bits:
+[Attestations: a new generation of signatures on PyPI].
+
+### How to view a file's attestations
+For consumers and package installers wanting to perform verification, PyPI
+currently provides two ways to access digital attestations associated with a
+given file on PyPI:
+
+* **A new Integrity API for PyPI**
+  The [Integrity API](https://docs.pypi.org/api/integrity/) provides programmatic
+  access to PyPI's implementation of PEP 740. Operating on individual files, it
+  collects all published attestations for a given file and returns them as a
+  single response.
+
+* **A new web UI for viewing per-file attributes**
+  Similarly, we have introduced a new page on PyPI's web UI, displaying details
+  for individual files, including the presence of any attestations about the
+  file. You can see an example here:
+  <https://pypi.org/project/sampleproject/#sampleproject-4.0.0.tar.gz>
+
+### Get started today
+
+The generation and publication of attestations happens by default, and no
+changes are necessary for projects that meet all of these conditions:
+
+1. publish from GitHub Actions;
+2. via [Trusted Publishing]; and
+3. use the [pypa/gh-action-pypi-publish] action to publish.
+
+Support for automatic attestation generation and publication from other Trusted
```

Comment:
While it's linked, should we explicitly mention that attestations can be generated from other CI providers that are supported by trusted publishing?

Reply:
Only GitHub is currently supported: https://docs.pypi.org/attestations/producing-attestations/#:~:text=Attestations%20are%20currently%20only%20supported%20when%20uploading%20with%20Trusted%20Publishing%2C%20and%20currently%20only%20with%20GitHub%2Dbased%20Trusted%20Publishers